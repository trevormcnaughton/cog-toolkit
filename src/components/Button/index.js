import React, { PureComponent } from 'react'
import theme from '../../util/theme'
import styled from 'styled-components'

export default class Button extends PureComponent {
  render() {
    const { className, children, disabled = false, primary, fit, style, small } = this.props
    const Button = styled.button`
      position: relative;
      display: inline-block;
      padding: 0.75rem 1.5rem;
      width: ${({ fit }) => (fit ? '100%' : 'auto')};
      color: ${({ primary }) => (primary ? '#FFF' : theme.color.primary)};
      font-family: var(--font-family-base);
      font-size: ${({ small }) => (small ? '0.75rem' : '0.875rem')};
      font-weight: 500;
      vertical-align: middle;
      white-space: nowrap;
      text-align: center;
      text-decoration: none;
      background: ${({ primary }) => (primary ? theme.color.primary : '#FFF')};
      border: 1px solid rgba(0, 0, 0, 0.08);
      border-radius: 2px;
      transition: background 200ms;
      outline: 0;
      overflow: hidden;
      cursor: pointer;
      appearance: none;
      user-select: none;
      transition: background-color 150ms, color 150ms, border 150ms,
        box-shadow 150ms;

      &:hover {
        background: ${({ primary }) =>
          primary ? theme.color.primaryLight : theme.color.greyLightest};
      }

      :disabled {
        pointer-events: none;
        cursor: not-allowed
        opacity: 0.7;
      }
    `

    return (
      <Button
        className={className}
        disabled={disabled}
        style={style}
        onClick={this.props.onClick}
        primary={primary}
        small={small}
        fit={fit}
      >
        {children}
      </Button>
    )
  }
}
