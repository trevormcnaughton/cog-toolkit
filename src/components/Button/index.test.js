import React from 'react'
import renderer from 'react-test-renderer'
import sinon from 'sinon'

import Button from './index'

const noop = () => {}

describe('Button', () => {
  describe('snapshots', () => {
    it('renders a default button with text', () => {
      const tree = renderer.create(
        <Button onClick={noop}>Button</Button>
      )

      expect(tree).toMatchSnapshot()
    })
    it('renders a primary button with text', () => {
      const tree = renderer.create(
        <Button primary onClick={noop}>Button</Button>
      )

      expect(tree).toMatchSnapshot()
    })
    it('renders a small default button with text', () => {
      const tree = renderer.create(
        <Button small onClick={noop}>Button</Button>
      )

      expect(tree).toMatchSnapshot()
    })
    it('renders a small primary button with text', () => {
      const tree = renderer.create(
        <Button primary small onClick={noop}>Button</Button>
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
