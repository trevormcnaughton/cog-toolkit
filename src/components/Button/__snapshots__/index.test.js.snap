// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots renders a default button with text 1`] = `
<button
  className="sc-bdVaJa esNCtN"
  onClick={[Function]}
  style={undefined}
>
  Button
</button>
`;

exports[`Button snapshots renders a primary button with text 1`] = `
<button
  className="sc-bwzfXH kgYHCc"
  onClick={[Function]}
  style={undefined}
>
  Button
</button>
`;

exports[`Button snapshots renders a small default button with text 1`] = `
<button
  className="sc-htpNat kHRfmE"
  onClick={[Function]}
  style={undefined}
>
  Button
</button>
`;

exports[`Button snapshots renders a small primary button with text 1`] = `
<button
  className="sc-bxivhb aYoIQ"
  onClick={[Function]}
  style={undefined}
>
  Button
</button>
`;
